src/lib/server/tools/hfDocs.ts

```diff
@@ -0,0 +1,47 @@
+import type { BackendTool } from ".";
+import { callSpace, getIpToken } from "./utils";
+
+const hfDocs: BackendTool = {
+	name: "hf_docs",
+	displayName: "Hugging Face Docs",
+	description:
+		"Use this tool to get relevant docs snippets regarding Hugging Face open source libraries (transformers, diffusers, accelerate, huggingface_hub) and Hugging Face hf.co platform.",
+	isOnByDefault: true,
+	parameterDefinitions: {
+		query: {
+			required: true,
+			type: "string",
+			description:
+				"A search query which will be used to fetch the most relevant docs snippets regarding the user's query",
+		},
+	},
+	async *call({ prompt }, { messages, ip, username }) {
+		const ipToken = await getIpToken(ip, username);
+
+		const userMessages = messages.filter(({ from }) => from === "user");
+		const previousUserMessages = userMessages.slice(0, -1);
+
+		const query =
+			(previousUserMessages.length
+				? `Previous questions: \n${previousUserMessages
+						.map(({ content }) => `- ${content}`)
+						.join("\n")}`
+				: "") +
+			"\n\nCurrent Question: " +
+			String(prompt);
```

Comment:
tried to match with https://github.com/huggingface/chat-ui/blob/900a9b395923aa8b6f9de849b1d41abd12f5c707/src/lib/server/websearch/search/generateQuery.ts#L53-L62